snapcraft/integrations/travis.py

```diff
@@ -0,0 +1,153 @@
+
+
+logger = logging.getLogger(__name__)
+
+
+def _encrypt_config(config_path):
+    """Encrypt given snapcraft config file for Travis jobs."""
+    cmd = [
+        'travis', 'encrypt-file',
+        '--force',
+        '--add', 'after_success',
+        '--decrypt-to', '.snapcraft.cfg',
+        config_path, '.travis_snapcraft.cfg',
+    ]
+    try:
+        subprocess.check_output(cmd, stderr=subprocess.PIPE)
+    except subprocess.CalledProcessError as err:
+        raise RuntimeError(
+            '`travis encrypt-file` failed: {}\n{}'.format(
+                err.returncode, err.stderr.decode()))
+
+
+@requires_command_success(
+    'travis version',
+    EnvironmentError(
+        'Travis CLI (`travis`) is not available.\n'
+        'Please install it (e.g. `sudo gem install travis`) '
+        'before trying this again.'),
+    EnvironmentError(
+        'Travis CLI (`travis version`) is not functional.\n'
+        'Make sure it works correctly in your system '
+        'before trying this again.'))
+@requires_command_success(
+    'git status',
+    EnvironmentError(
+        'Git (`git`) is not available, this tool cannot verify '
+        'its prerequisites.\n'
+        'Please install it (e.g. `sudo apt install git`) '
+        'before trying this again.'),
+    EnvironmentError(
+        'The current directory is not a Git repository.\n'
+        'Please switch to the desired project repository where '
+        'Travis should be enabled.'))
+@requires_path_exists(
+    '.travis.yml',
+    'Travis project is not initialised for the current directory.\n'
+    'Please initialise Travis project (e.g. `travis init`) with '
+    'appropriate parameters.')
+def enable():
+    series = storeapi.constants.DEFAULT_SERIES
+    project_config = load_config()
+    snap_name = project_config.data['name']
+    logger.info(
+        'Enabling Travis testbeds to push and release "{}" snaps '
+        'to edge channel in series {}'.format(snap_name, series)
+    )
+
+    packages = [{'name': snap_name, 'series': series}]
+    channels = ['edge']
+
+    # XXX cprov 20161116: Needs caveat syntax for restricting
+    # origins (IP or reverse-dns) but Travis sudo-enabled containers
+    # do not have static egress routes.
+    # See https://docs.travis-ci.com/user/ip-addresses.
+
+    logger.info('Acquiring specific authorization information ...')
+    store = storeapi.StoreClient()
+    if not _login(store, packages=packages, channels=channels, save=False):
+        raise RuntimeError(
+            'Cannot continue without logging in successfully.')
+
+    logger.info(
+        'Encrypting authorization for Travis and adjusting project to '
+        'automatically decrypt and use it during "after_success".')
+
+    with tempfile.NamedTemporaryFile(mode='w') as fd:
+        store.conf.parser.write(fd)
+        fd.flush()
+        _encrypt_config(fd.name)
+
+    with open('.travis.yml', 'r+') as fd:
+        travis_conf = yaml.load(fd)
```

Comment:
this is taking advantage of our `OrderedDict` right? well I am more interested in the `dump`